templates/rule.conf.erb

```diff
@@ -1,4 +1,8 @@
+# File controlled by Puppet
+# DO NOT EDIT LOCALLY
+#
 # <domain> <type> <item> <value>
 #
-
-<%= @ulimit_domain %>		<%= @ulimit_type %>		<%= @ulimit_item %>		<%= @ulimit_value %>
+<% (0..@ulimit_type.length-1).each do |i| %>
```

Comment:
I did it this way because each parameter is an array and as the domain needs to have its values to the right, doing an array iteration won't work as expecting and can cause formatting issues. By doing this for loop it allows control over the positioning, whilst inputting the array number at that specific point. 

In the rule.pp file there is a check to ensure that all the array's are the same size, before processing, so that length is consistent with the all the parameters